orangecontrib/prototypes/widgets/owdatasets.py

```diff
@@ -0,0 +1,514 @@
+            openExternalLinks=True,
+            textInteractionFlags=Qt.TextSelectableByMouse |
+                                 Qt.LinksAccessibleByMouse
+        )
+        self.descriptionlabel.setFrameStyle(QTextBrowser.NoFrame)
+        # no (white) text background
+        self.descriptionlabel.viewport().setAutoFillBackground(False)
+
+        box.layout().addWidget(self.descriptionlabel)
+        self.splitter.addWidget(self.view)
+        self.splitter.addWidget(box)
+
+        self.splitter.setSizes([300, 200])
+        self.splitter.splitterMoved.connect(
+            lambda:
+                setattr(self, "splitter_state",
+                        bytes(self.splitter.saveState()))
+        )
+        self.mainArea.layout().addWidget(self.splitter)
+        self.controlArea.layout().addStretch(10)
+        gui.auto_commit(self.controlArea, self, "auto_commit", "Commit")
+
+        model = QStandardItemModel(self)
+        model.setHorizontalHeaderLabels(HEADER)
+        proxy = QSortFilterProxyModel()
+        proxy.setSourceModel(model)
+        self.view.setModel(model)
+
+        if self.splitter_state:
+            self.splitter.restoreState(self.splitter_state)
+
+        self.view.setItemDelegateForColumn(
+            Header.Size, SizeDelegate(self))
+        self.view.setItemDelegateForColumn(
+            Header.Local, gui.IndicatorItemDelegate(self, role=Qt.DisplayRole))
+        self.view.setItemDelegateForColumn(
+            Header.Instances, NumericalDelegate(self))
+        self.view.setItemDelegateForColumn(
+            Header.Variables, NumericalDelegate(self))
+
+        self.view.resizeColumnToContents(Header.Local)
+
+        if self.header_state:
+            self.view.header().restoreState(self.header_state)
+
+        self.setBlocking(True)
+        self.setStatusMessage("Initializing")
+
+        self._executor = ThreadPoolExecutor(max_workers=1)
+        f = self._executor.submit(list_remote)
+        w = FutureWatcher(f, parent=self)
+        w.done.connect(self.__set_index)
+
+    @Slot(object)
+    def __set_index(self, f):
+        # type: (Future) -> None
+        # set results from `list_remote` query.
+        assert QThread.currentThread() is self.thread()
+        assert f.done()
+        self.setBlocking(False)
+        self.setStatusMessage("")
+        allinfolocal = list_local()
+        try:
+            res = f.result()
+        except Exception as er:
+            log = logging.getLogger(__name__)
+            log.exception("Error while fetching updated index")
+            exc_text = format_exception(er)
+            if not allinfolocal:
+                self.error("Could not fetch data set list: {}"
+                           .format(exc_text))
+            else:
+                self.warning("Could not fetch data sets list, only local "
+                             "cached data sets are shown ({})"
+                             .format(exc_text))
```

Comment:
I think the message --- _without_ the exception --- should be enough (for now), since this also appears during a "normal" workload (no internet).